README.md
# Extension dataquery

SQL-based data consumer for the Tesseract project.
Please refer to the [manual for more information](https://docs.typo3.org/typo3cms/extensions/dataquery/).

ChangeLog
2019-06-04 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Add missing workspsace-related field, resolves #1
	* Release version 2.1.3

2019-02-20 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Released to TER as version 2.1.2

2018-11-23 Julien Henchoz (Cobweb)  <typo3@cobweb.ch>

	* Replaced deprecated calls

2018-07-17 Julien Henchoz (Cobweb)  <typo3@cobweb.ch>

	* Fixed query alias replacement

2017-06-06 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Released to TER as version 2.1.0
	* Released to TER as version 2.1.1

2017-05-26 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Verified compatibility with TYPO3 CMS 8 LTS

2016-04-22 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Enhanced fulltext search capabilities, resolves #75041
	* Released to TER as version 2.0.0, resolves #75865

2016-04-17 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Fixed failing unit tests, resolves #75722

2016-04-15 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Applied PSR-2 formatting and other cleanups, resolves #75678
	* Moved language files to XLIFF, resolves #75685
	* Changed default query cache duration to 0, resolves #57138

2016-03-14 Fabien Udriot  <fabien.udriot@ecodev.ch>

	* Added composer file, resolves #75051
	* Ensured compatibility with TYPO3 CMS 7 LTS, resolves #75052

2015-09-27 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Fixed crashing query check wizard, resolves #70136
	* Ensured compatibility with TYPO3 CMS 7.5, resolves #70137
	* Added reference to description field, resolves #70139

2015-06-09 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Moved to namespaces, verified compatibility with TYPO3 CMS 7, resolves #67370

2015-02-27 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Fixed wrong output in custom fulltext field, resolves #65409
	* Released to TER as version 1.11.1, resolves #65410

2015-02-11 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Released to TER as version 1.11.0, resolves #65008

2015-02-09 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Added fulltext search capability, resolves #41882

2015-02-01 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Refactored existing unit tests, resolves #64488

2015-01-25 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Moved documentation to reStructuredText, resolves #64487

2014-07-22 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Improved query rebuilding log message, resolves #60461

2014-05-09 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Compatibility verified with TYPO3 CMS 6.2, resolves #58603
	* Added hook to cache hash calculation process, resolves #58601
	* Released to TER as version 1.10.0, resolves #58604

2014-03-21 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Ensured preservation of localized uid information in data structure, resolves #57155

2013-10-15 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Adjusted SQL definition to fit TYPO3 Install Tool, resolves #51967

2013-06-21 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Released to TER as version 1.9.0

2013-06-06 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Added index on cache table for improved performance, resolves #48834

2013-05-22 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Corrected SQL definition of cache table, resolves #48454
	* Compatibility verified with TYPO3 CMS 6.1, resolves #47517

2013-04-11 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Solved issue with trimming beyond query parts length, resolves #42802

2013-04-01 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Refactored SQL parser test suite, resolves #46846
	* Added test cases for SQL parsing exceptions, resolves #46847
	* Compatibility verified with TYPO3 CMS 6.0, resolves #46851

2012-12-10 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Fixed abusive overriding of parent method in tx_dataquery_wrapper, resolves #43843
	* Released to TER as version 1.8.2

2012-09-10 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Fixed error with natural sorting in PHP-side sorting algorithm, resolves #40655
	* Released to TER as version 1.8.1

2012-07-30 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Released to TER as version 1.8.0

2012-07-24 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Added handling of "engine" property in ordering, resolves #39193

2012-07-10 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Released to TER as version 1.7.0

2012-06-29 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Added definition of search fields, resolves #38440

2012-06-26 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Replaced all t3lib_div::debug() calls, references #38236

2012-06-21 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Added note in the documentation about the use of brackets in queries, resolves #37151

2012-06-20 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Verified compatibility with TYPO3 4.7, raised requirements to TYPO3 4.5, resolves #38236
	* Cleaned up unused bits of code, completed phpDoc comments
	* Added note in the documentation about static values not being supported in queries, resolves #37179

2012-04-19 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Released to TER as version 1.6.0

2012-04-03 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Fixed irregular malfunctions of the validation wizard, resolves #35594
	* Added warning about duplicate fields in query, resolves #33061
	* Added support for random ordering from data filter, references #27780

2012-04-02 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Added support for special ordering keywords (RAND, NULL), resolves #27780

2012-03-20 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Updated the process of initializing the empty data structure, resolves #35050

2012-02-06 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Released to TER as version 1.5.0

2012-02-03 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Added possibility to return also id-list type data structures, resolves #33602

2012-01-19 Francois Suter (Cobweb)  <typo3@cobweb.ch>

	* Externalized transformation of data filter conditions to SQL, resolves #33317

ext_emconf.php
<?php

/***************************************************************
 * Extension Manager/Repository config file for ext "dataquery".
 *
 * Auto generated 06-06-2017 17:12
 *
 * Manual updates:
 * Only the data in the array - everything else is removed by next
 * writing. "version" and "dependencies" must not be touched!
 ***************************************************************/

$EM_CONF[$_EXTKEY] = [
        'title' => 'SQL-based Data Provider - Tesseract project',
        'description' => 'Assembles a query on data stored in the TYPO3 CMS local database, automatically enforcing criteria like language, publication date, etc. More info on http://www.typo3-tesseract.com/',
        'category' => 'misc',
        'author' => 'Francois Suter (Cobweb)',
        'author_email' => 'typo3@cobweb.ch',
        'state' => 'stable',
        'uploadfolder' => 0,
        'createDirs' => '',
        'clearCacheOnLoad' => 0,
        'author_company' => '',
        'version' => '2.1.3',
        'constraints' =>
                [
                        'depends' =>
                                [
                                        'typo3' => '7.6.0-8.99.99',
                                        'tesseract' => '2.0.0-0.0.0',
                                        'datafilter' => '2.0.0-0.0.0',
                                        'overlays' => '3.0.0-0.0.0',
                                        'expressions' => '2.0.0-0.0.0',
                                ],
                        'conflicts' =>
                                [
                                ],
                        'suggests' =>
                                [
                                ],
                ],
];

